src/portal/lib/nodes.ts

```diff
@@ -258,6 +258,15 @@ export async function getNodeUsedResources(nodeId: string) {
   }
 }
 
+export async function getFarmDetails(farmID: string) {
+  const res = await axios.get(`${config.gridproxyUrl}/farms?farm_id=${farmID}`, {
+    timeout: 5000,
+  });
+
+  if (res.status === 200) {
+    return res.data;
+  } else throw new Error("Unexpected error while fetching farm data: " + res.data);
```

Comment:
It'll be better if you returned the message of the request instead.

Reply:
should be `res.data` only?

Reply:
idk, print the response and track where the error message.

Reply:
I thought it depends on the server, some servers return the error message on the `res.data`, I will investigate on it :D